core/src/main/scala/org/apache/spark/ui/jobs/StageTable.scala

```diff
@@ -417,19 +417,8 @@ private[ui] class StageDataSource(
       case Some(t) => UIUtils.formatDate(t)
       case None => "Unknown"
     }
-    val finishTime = stageData.completionTime.map(_.getTime()).getOrElse(currentTime)
-
-    // The submission time for a stage is misleading because it counts the time
-    // the stage waits to be launched. (SPARK-10930)
-    val duration = stageData.firstTaskLaunchedTime.map { date =>
-      val time = date.getTime()
-      if (finishTime > time) {
-        finishTime - time
-      } else {
-        None
-        currentTime - time
-      }
-    }
+
+    val duration = Some(stageData.executorRunTime)
```

Comment:
Just to be clear, this is exactly the field used to compute min/max stats, etc?

Reply:
I think no. `stageData.executorRunTime` represents summation of `executorRunTime` of every task corresponding to the stage. Tasks need not be execute in sequential order (can be parallel also). So, I think this might not give the correct duration.

I think, field used to compute min/max stats, etc are `taskMetrics.executorRunTime`. @teeyog please correct me if I am wrong.

Reply:
> I think no. `stageData.executorRunTime` represents summation of `executorRunTime` of every task corresponding to the stage. Tasks need not be execute in sequential order (can be parallel also). So, I think this might not give the correct duration.
> 
> I think, field used to compute min/max stats, etc are `taskMetrics.executorRunTime`. @teeyog please correct me if I am wrong.

What you said is correct. When the task is executed in parallel, the result is wrong. Then I think the task duration should not use executorRunTime. As shown in the previous figure, their values ​​are not match.

Reply:
Do you think I am right, if I am going to develop

Reply:
Thanks @teeyog. I think this was the behavior in the previous versions of spark. I am not sure we need to change it. What ever you are proposing is `duration = executorRunTime + ser/deser time + gettingResultTime`. But, Already these 3 columns are getting disaplyed in the task table right? If we unify all would we need to display individual columns?

Reply:
@shahidki31 I think the task duration should be calculated using the ```completed time - launched time``` instead of the executorRunTime. This ensures that a stage has only one task, and the stage duration and task duration are the same.

Reply:
Yes. `completed time - launched time = executorRunTime + ser/deser time + gettingResultTime`. right? Also, I think these 3 columns are getting displayed in the task table, Right?. If you really want, you may can add a new column and you can name it  `totalTime` which display the summation of 3. But still, I am not sure that will match with stage duration, as stage may can have other parameters for duration.

Reply:
@shahidki31 
Yes, I think this can match with stage duration